src/Microsoft.OData.Edm/Schema/EdmReferentialConstraint.cs

```diff
@@ -24,7 +24,7 @@ public class EdmReferentialConstraint : IEdmReferentialConstraint
         public EdmReferentialConstraint(IEnumerable<EdmReferentialConstraintPropertyPair> propertyPairs)
         {
             EdmUtil.CheckArgumentNull(propertyPairs, "propertyPairs");
-            this.propertyPairs = propertyPairs.ToList();
+            this.propertyPairs = propertyPairs;

```

Comment:
this changes semantics slightly by delaying iteration until the caller calls PropertyPairs.  Are we okay with that change in semantics?  Are we certain we will be able to iterate propertyPairs multiple times after the referential constraint is constructed?